app/assets/javascripts/gift_requests.js

```diff
@@ -0,0 +1,45 @@
+$(function() {
+
+  if ($('#gift_request_link').val()) {
+    fetchAmazonItemName();
+  }
+
+  $('#gift_request_link').on('input', function(e) {
+    fetchAmazonItemName();
+  });
+
+  function fetchAmazonItemName() {
+    var amazonInfo = $('#amazon-info');
+    amazonInfo.html('');
+
+    $.ajax({
+      url: '/gift_requests/fetch_amazon_info',
```

Comment:
you can just put `.json` on the end of this and rails will auto-interpret the content type